pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -49,15 +52,20 @@ func (d *DualWriterMode1) Create(ctx context.Context, original runtime.Object, c
 	}
 	d.recordLegacyDuration(false, mode1Str, options.Kind, method, startLegacy)
 
+	d.pendingActions.Add(1)
 	go func() {
+		defer d.pendingActions.Done()
+		ctx := context.WithoutCancel(ctx)
 		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage create timeout"))
+		defer cancel()
```

Comment:
How the `defer` statement works:

- https://gobyexample.com/defer
- https://go.dev/tour/flowcontrol/12